### ISSUE TITLE
Merlin server failed to build w/ Go 1.15 due to supporting quic-go package

### ISSUE THREAD

**Issue by g3rzi** [Ne0nd0g/merlin#99]
### Prerequisite
* [x] I have read the [README](https://github.com/Ne0nd0g/merlin/blob/master/README.MD)
* [x] I have search the opened & _closed_ [issues](https://github.com/Ne0nd0g/merlin/issues)
* [x] I have search the [WIKI](https://github.com/Ne0nd0g/merlin/wiki) and its [FAQ](https://github.com/Ne0nd0g/merlin/wiki/FAQ) page
### Environment Data
* Merlin Version: 0.9.1-beta (from master)
* Merlin Build:  nonRelease (from master)
* Go Version: go version go1.15 linux/amd64
* GOPATH Environment Variable:  /home/ubuntu/go (result of `go env GOPATH`)
* GOROOT Environment Variable: /usr/local/go (result of `go env GOROOT`)
* Operating System:  Linux ubuntu 5.3.0-1032-aws #34~18.04.2-Ubuntu SMP Fri Jul 24 10:06:28 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

### Expected Behavior
Merlin server should start run after building

### Actual Behavior
After building it fail to run with the following error:  
```
root@ubuntu:~/merlin# ./main
panic: qtls.ClientSessionState not compatible with tls.ClientSessionState

goroutine 1 [running]:
github.com/lucas-clemente/quic-go/internal/handshake.init.2()
        /home/ubuntu/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.17.3/internal/handshake/unsafe.go:26 +0x205
```

### Steps to Reproduce Behavior
```
git clone https://github.com/Ne0nd0g/merlin.git
cd merlin/
go build cmd/merlinserver/main.go
./main
```

After running the main it fails.

### Misc Information
On different machine I have Go version `go1.14.4` and I don't have this problem.   
Not sure it is related to the version number but it might be.

**Comment by Ne0nd0g** [comment]
What version of Go did you try to use the first time? You will need Go Module support enabled.

**Comment by Ne0nd0g** [comment]
Oh, I see. Go 1.15 isn't supported because `quick-go` doesn't support it.

**Comment by g3rzi** [comment]
They wrote on their [website](https://github.com/lucas-clemente/quic-go):

> We currently support Go 1.14+, with Go modules support enabled.  

I thought it should support also `1.15`.

**Comment by cdf123** [comment]
Changing go.mod to pull in "quic-go v0.18.0" seemed to fix the issue for me.  It's possible newer versions could work as well.

**Comment by Pusher91** [comment]
> 
> 
> Changing go.mod to pull in "quic-go v0.18.0" seemed to fix the issue for me. It's possible newer versions could work as well.

How do you do this?  Sorry I'm new to using go.  I've started reading about go modules online to figure this out but it is a lot to take in.

**Comment by Ne0nd0g** [comment]
You would change the version number on line 23 to match the tagged version you want to use

https://github.com/Ne0nd0g/merlin/blob/master/go.mod#L23

**Comment by Ne0nd0g** [comment]
Fixed in v1.0.0 that upgraded to Go 1.16 and quic-go v0.20.1
